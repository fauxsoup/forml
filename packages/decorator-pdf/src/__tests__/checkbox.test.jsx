import Checkbox from '../checkbox';
import Context from '@forml/context';
import React from 'react';
import ReactPDF from '@react-pdf/renderer';
import { render } from '@testing-library/react';
import * as decorator from '../';

describe('renders', function () {
    let form;
    let schema;
    let title = 'title';
    let description = 'description';
    let styles;
    let root;
    let header;
    let icon;

    beforeEach(function () {
        root = jest.fn(() => ({ border: '1px solid red' }));
        header = jest.fn(() => ({ border: '1px solid red' }));
        icon = jest.fn(() => ({ border: '1px solid red' }));
        schema = { type: 'boolean' };
        styles = {
            get root() {
                return root();
            },
            get header() {
                return header();
            },
            get icon() {
                return icon();
            },
        };
        form = { type: 'checkbox', key: [] };
    });

    test('with no title or description', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Checkbox form={form} />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('with title but no description', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Checkbox form={form} title={title} />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('when checked', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Checkbox
                        form={form}
                        title={title}
                        description={description}
                        checked={true}
                    />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('with optional formStyles', function () {
        form = { type: 'checkbox', styles, key: [] };

        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Checkbox
                        form={form}
                        title={title}
                        description={description}
                    />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
        expect(root).toHaveBeenCalled();
        expect(header).toHaveBeenCalled();
        expect(icon).toHaveBeenCalled();
    });
});
