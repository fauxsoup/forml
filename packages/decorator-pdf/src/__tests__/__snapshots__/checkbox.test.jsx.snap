// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when checked 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex-direction: column; flex: 1;"
    >
      <view
        style="display: flex; flex-direction: row; align-items: center;"
      >
        <view
          style="display: inline-flex; font-family: Material Icons; font-size: 16pt; padding: 4pt 0px; min-width: 20pt;"
        >
          <text>
            check_box
          </text>
        </view>
        <text
          style="font-size: 10pt; font-weight: bold; padding-top: 4pt; margin-bottom: 2pt;"
        >
          title
        </text>
      </view>
      <view
        style="margin-top: 2pt; font-size: 9pt; color: rgba(0, 0, 0, 0.54);"
      >
        <text>
          description
        </text>
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with no title or description 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex-direction: column; flex: 1;"
    >
      <view
        style="display: flex; flex-direction: row; align-items: center;"
      >
        <view
          style="display: inline-flex; font-family: Material Icons; font-size: 16pt; padding: 4pt 0px; min-width: 20pt;"
        >
          <text>
            check_box_outline_blank
          </text>
        </view>
        <text
          style="font-size: 10pt; font-weight: bold; padding-top: 4pt; margin-bottom: 2pt;"
        />
      </view>
      <view
        style="margin-top: 2pt; font-size: 9pt; color: rgba(0, 0, 0, 0.54);"
      >
        <text />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with optional formStyles 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex-direction: column; flex: 1; border: 1px solid red;"
    >
      <view
        style="display: flex; flex-direction: row; align-items: center; border: 1px solid red;"
      >
        <view
          style="display: inline-flex; font-family: Material Icons; font-size: 16pt; padding: 4pt 0px; min-width: 20pt; border: 1px solid red;"
        >
          <text>
            check_box_outline_blank
          </text>
        </view>
        <text
          style="font-size: 10pt; font-weight: bold; padding-top: 4pt; margin-bottom: 2pt;"
        >
          title
        </text>
      </view>
      <view
        style="margin-top: 2pt; font-size: 9pt; color: rgba(0, 0, 0, 0.54);"
      >
        <text>
          description
        </text>
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with title but no description 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex-direction: column; flex: 1;"
    >
      <view
        style="display: flex; flex-direction: row; align-items: center;"
      >
        <view
          style="display: inline-flex; font-family: Material Icons; font-size: 16pt; padding: 4pt 0px; min-width: 20pt;"
        >
          <text>
            check_box_outline_blank
          </text>
        </view>
        <text
          style="font-size: 10pt; font-weight: bold; padding-top: 4pt; margin-bottom: 2pt;"
        >
          title
        </text>
      </view>
      <view
        style="margin-top: 2pt; font-size: 9pt; color: rgba(0, 0, 0, 0.54);"
      >
        <text />
      </view>
    </view>
  </page>
</div>
`;
